sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -152,6 +155,25 @@ export class AssessmentControllerService {
     return disbursementDetails;
   }
 
+  /**
+   * Calculate the sum of value amount for each disbursement value within the disbursement schedules.
+   * @param disbursementSchedules disbursement schedule details.
+   * @returns The total sum of value amount across all disbursement schedules.
+   */
+  private sumDisbursementValueAmounts(
+    disbursementSchedules: DisbursementSchedule[],
+  ): number {
+    return disbursementSchedules
+      .flatMap(
+        (disbursementSchedule) => disbursementSchedule.disbursementValues,
+      )
+      .reduce(
+        (accumulator, disbursementValue) =>
+          accumulator + disbursementValue.valueAmount,
+        0,
```

Comment:
IMO: Specifying `0` as the initial value explicitly to the `reduce` is not required here.

Reply:
Creating an accumulator in the start, in case if the array is null, then the value returned will be 0.